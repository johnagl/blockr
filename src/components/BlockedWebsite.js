import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';
import DeleteIcon from '@material-ui/icons/Delete';

const BlockedWebsite = ( {labelId, value, handleRemoveWebsite } ) => {
    return (
        <ListItem fullWidth key={value} role={undefined} button>
            <ListItemText id={labelId} primary={value} />
                <ListItemSecondaryAction>
                    <IconButton edge="end" onClick={() => handleRemoveWebsite(value)}>
                        <DeleteIcon />
                            </IconButton>
                </ListItemSecondaryAction>
        </ListItem>
    )
}

export default BlockedWebsite;