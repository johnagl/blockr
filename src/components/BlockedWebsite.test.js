import { shallow } from "enzyme";
import React from 'react';
import BlockedWebsite from './BlockedWebsite';

it('expect to render blocked website list', () => {
    expect(shallow(<BlockedWebsite/>)).toMatchSnapshot();
})