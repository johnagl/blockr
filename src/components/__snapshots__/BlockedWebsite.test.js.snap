// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render blocked website list 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  fullWidth={true}
>
  <WithStyles(ForwardRef(ListItemText)) />
  <WithStyles(ForwardRef(ListItemSecondaryAction))>
    <WithStyles(ForwardRef(IconButton))
      edge="end"
      onClick={[Function]}
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(ListItemSecondaryAction))>
</WithStyles(ForwardRef(ListItem))>
`;
